src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalLocaleOpt.java

```diff
@@ -42,27 +57,98 @@
 	    	if (uri.getScheme().equals("jar")) {
 	    		try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object>emptyMap())) {
 	    			path = fileSystem.getPath(path.toString());
-	    			getSupportedLocales(path, supportedLocales, displayLanguage);
+	    			getSupportedLocales(path, supportedLocales, languagesNames);
 	    		}
 			} else {
 				path = Paths.get(uri);
-				getSupportedLocales(path, supportedLocales, displayLanguage);
+				getSupportedLocales(path, supportedLocales, languagesNames);
 			}
 	    	
 		} catch (Exception e) {
 			logger.debug(e.getMessage());
 		}
     	return supportedLocales;
     }
-    
-    private void getSupportedLocales(Path path, Map<String, String> supportedLocales, String displayLanguage) throws IOException {
-    	Locale inLocale = Locale.forLanguageTag(displayLanguage); 
+
+    private void getSupportedLocales(Path path, Map<String, String> supportedLocales, JSONObject languagesNames) throws IOException {
     	try (Stream<Path> listOfFiles = Files.walk(path).filter(p -> Files.isRegularFile(p))) {
         	listOfFiles.map(file -> {
 				String fileName = file.getFileName().toString();
-				return fileName.substring(BUNDLE_PREFIX.length(), fileName.indexOf('.'));
-			}).forEach(s->supportedLocales.put(s, Locale.forLanguageTag(s).getDisplayName(inLocale)));
+				if(fileName.startsWith(BUNDLE_PREFIX) && fileName.endsWith(BUNDLE_SUFFIX)) {
+					return fileName.substring(BUNDLE_PREFIX.length(), fileName.indexOf('.'));
+				}
+				return "";
+			}).forEach(language -> {
+				if(language != null && !"".equalsIgnoreCase(language) && !ConstantsKeys.SOURCE.equalsIgnoreCase(language)) {
+					if (languagesNames != null) {
+						supportedLocales.put(language, (String) languagesNames.get(language));
+					} else {
+						supportedLocales.put(language, "");
+					}
+				}
+			});
         }
     }
 
+    private JSONObject getLanguagesNames(String displayLanguage){
```

Comment:
Yes, because in online mode Singleton service get the language names from CLDR, so for offline mode I change the code to get the language names from CLDR, too.